### PR TITLE
remove initializer for setting identifiers in events

### PR DESCRIPTION
**Pull request by ndurell**
# Description

We no longer want to allow setting identifiers, time, and profile in events. This can create confusing bugs for developers and brings it in line with the Android SDK.

# Check List

- [x] Are you changing anything with the public API?
- [ ] Have you tested this change on real device?
- [ ] Are your changes backwards compatible with previous SDK Versions?
- [ ] Have you added unit test coverage for your changes?
- [ ] Have you verified that your changes are compatible with all the operating system version this SDK currently supports?

# Manual Test Plan

1.

# Supporting Materials

### CODE REVIEW

**Comment by ndurell** [comment]
This will be released as a minor point release first and then removed in upcoming 3.0.0.

File: Sources/KlaviyoSwift/KlaviyoModels.swift [hunk 1]
```diff
@@ -63,6 +63,7 @@ public struct Event: Equatable {
     public let uniqueId: String
     public let identifiers: Identifiers?
 
+    @available(*, deprecated, renamed: "init(name:properties:value:uniqueId:)", message: "This initializer has been deprecated. Setting profile properties should be set prior to logging events.")
```
Comment (ajaysubra) [inline Sources/KlaviyoSwift/KlaviyoModels.swift:66]:
```suggestion
    @available(*, deprecated, renamed: "init(name:properties:value:uniqueId:)", message: "This initializer has been deprecated. Profile properties should be set prior to logging events.")
```

**Review by ajaysubra — APPROVED** [review]

File: Sources/KlaviyoSwift/KlaviyoModels.swift [hunk 2]
```diff
@@ -87,6 +88,19 @@ public struct Event: Equatable {
...
+    public init(name: EventName,
+                properties: [String: Any]? = nil,
+                value: Double? = nil,
+                uniqueId: String? = nil) {
+        metric = .init(name: name)
+        _profile = AnyCodable([:])
+        _properties = AnyCodable(properties ?? [:])
+        identifiers = nil
+        self.value = value
+        time = environment.analytics.date()
```
Comment (kennyklaviyo) [inline Sources/KlaviyoSwift/KlaviyoModels.swift:101]: shouldn't this be `self.identifiers` and `self.time`?

Comment (ndurell) [reply Sources/KlaviyoSwift/KlaviyoModels.swift:101]: Not sure I'm totally following but you don't have to use self if it's clear what you are setting. I think I have a swiftlint rule that will remove this but I can try setting it.

**Review by evan-masseau — APPROVED** [review]

**Pull request merged**